src/path/dec_int.rs

```diff
@@ -55,6 +55,17 @@ impl DecInt {
     }
 }
 
+impl core::fmt::Write for DecInt {
```

Comment:
If I understand correctly, this impl is visible in the public API; it's untidy to have public things we don't want users to use, and we'd have to think about whether `self.len + s.len()` could overflow. Would it work to declare a new private type which implements `fmt::Write`, and is wrapped by `DecInt`?

Reply:
Done